### PR TITLE
add skeleton loader for projects page

### LINKED ISSUE #149: [FEATURE] add skeleton loader for the projects page
### Description

Requirement:
- Add skeleton loader to the projects page, so when the data is not loaded 100% it will show skeleton laoding.

### PR DESCRIPTION
**Pull request by JakubChorzepa**
### Fixes Issue

Closes #149 

### Screenshots
Project page behaviour:
Before
Before

After:
After

### CODE REVIEW

**Comment by JakubChorzepa** [comment]
Sorry that i added these two more commits but my changes differed from #158 at that two lines and I just corrected them

File: components/Projects/ProjectsList.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  // Generate array of specified length with random key value
+  const skeletonProjectsToLoad = Array.from({length: 10}, () => (Math.random() + 1).toString(36).substring(7));
+
+  useEffect(() => {
+    if(data) {
+      setTimeout(() => {
+        setIsLoading(false)
+      }, 1000);
+    }
```
Comment (rohitdasu) [inline components/Projects/ProjectsList.tsx:20]: hi @JakubChorzepa could you please tell me why 1 sec delay for this? just curious

**Comment by rohitdasu** [comment]
@JakubChorzepa in my screen i'm seeing that skeleton image placeholder is having more height than the real project card. could you fix this issue? thanks 😊

File: components/Projects/ProjectsList.tsx (hunk 1 above)
Comment (JakubChorzepa) [reply components/Projects/ProjectsList.tsx:20]: I just wanted to avoid quick flick on page when data would be load fast

**Comment by leewan-09** [comment]
We need to have consistent card sizes

Untitled design

**Comment by JakubChorzepa** [comment]
I didn't set exact same height becoause actual setting fixed size will be only for this specific since the image on Projects.tsx has only specified width value but height i changing depending on image aspect ratio

**Comment by rohitdasu** [comment]
> I didn't set exact same height becoause actual setting fixed size will be only for this specific image since the image on Projects.tsx has only specified width value but height i changing depending on image aspect ratio

what's the solution for it @JakubChorzepa ?

**Comment by JakubChorzepa** [comment]
I think would be setting fixed size on image wrapper on projekt card and setting image to have full width and height and to object fit on image. Then to copy this behavior to skeleton component

**Comment by leewan-09** [comment]
For me, color is much more bright. 

Something like this is much nicer 
image

https://flowbite.com/docs/components/skeleton/#card-placeholder

**Comment by rohitdasu** [comment]
> I think would be setting fixed size on image wrapper on project card and setting image to have full width and height and to object fit on image. Then to copy this behavior to skeleton component

yes make sense. could you try please it?

**Comment by JakubChorzepa** [comment]
> For me, color is much more bright.
> 
> Something like this is much nicer image
> 
> https://flowbite.com/docs/components/skeleton/#card-placeholder

Yeah i used that page and i dont know why I changed that color 😄, will get it to this one

**Comment by JakubChorzepa** [comment]
> > I think would be setting fixed size on image wrapper on project card and setting image to have full width and height and to object fit on image. Then to copy this behavior to skeleton component
> 
> yes make sense. could you try please it?

Will do that changes tomorrow, I mean on european time becoause I don't have access to computer rn

**Comment by rohitdasu** [comment]
Sure ☑️

**Comment by leewan-09** [comment]
Also, the number line appears to be excessive. With some padding, 4 or 5 lines will look ideal. What do you say?
image

**Comment by JakubChorzepa** [comment]
Yeah 4 lines with padding will look better

**Comment by leewan-09** [comment]
@JakubChorzepa resolve conflicts

**Comment by leewan-09** [comment]
The light mode color is perfect, In Dark mode, I think still it little too bright.

**Comment by JakubChorzepa** [comment]
Ok I'll change it

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (leewan-09) [inline .vscode/settings.json:2]: could you remove this file

**Review by leewan-09 — APPROVED** [review]
LGTM

**Review by rohitdasu — APPROVED** [review]

**Pull request merged**